cram/cram_encode.c

```diff
@@ -1822,6 +1829,16 @@ int cram_encode_container(cram_fd *fd, cram_container *c) {
         c->ref_seq_id = c->ref_id;
     }
 
+    if (!no_ref && c->refs_used) {
+        for (i = 0; i < nref; i++) {
+            if (c->refs_used[i]) {
+                cram_get_ref(fd, i, 1, 0);
```

Comment:
It was going well until I enabled multi-threading:
```
REF_PATH=/tmp/cc/%s ./test/test_view -C -@ 8 -p /tmp/t8.cram /tmp/9305_4#1n_nour.bam
[W::cram_get_ref] Failed to populate reference for id 5
[W::cram_encode_container] Failed to load reference #5
[W::cram_encode_container] Enabling embed_ref=2 mode to auto-generate reference
[W::cram_encode_container] NOTE: the CRAM file will be bigger than using an external reference
[W::cram_get_ref] Failed to populate reference for id 16
[W::cram_encode_container] Failed to load reference #16
[W::cram_encode_container] Switching to non-ref mode
[W::cram_get_ref] Failed to populate reference for id 1
[E::validate_md5] SQ header M5 tag discrepancy for reference '10'
[E::validate_md5] Please use the correct reference, or consider using embed_ref=2
[E::cram_flush_thread] Call to cram_encode_container failed
Error writing output.
```

Note that you might have to run it a few times to get this to happen.  I think it depends on the main thread managing to queue up enough containers before one of the workers switches it to `no_ref` mode.

I suspect the easiest solution might be to check for reference loading failures here, and switch the container to `no_ref` mode if it happens.

Reply:
> Note that you might have to run it a few times to get this to happen. I think it depends on the main thread managing to queue up enough containers before one of the workers switches it to `no_ref` mode.
> 
> I suspect the easiest solution might be to check for reference loading failures here, and switch the container to `no_ref` mode if it happens.

More than a few times for me - it was failing about 1 in 100 on my smaller test set (with approx 10 containers worth and 3 threads).  Anyway, it's now doing as suggested and I can't break it any more (several thousand runs done).